org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/ProcessInfoDialog.java

```diff
@@ -216,6 +245,64 @@ public void onEvent(Event event) throws Exception {
 		if (event.getTarget() == btnOk) {
 			this.detach();
 		}
+		if(event.getTarget() == btnPrint) {
+			Clients.showBusy("Processing...");
+			Clients.response(new AuEcho(ProcessInfoDialog.this, "onPrint", null));
+		}
+	}
+	
+	/**
+	 * On Print
+	 */
+	public void onPrint() {
+		// Loop through all items
+		List<File> pdfList = new ArrayList<File>();
+		for (int i = 0; i < m_logs.length; i++)
+		{
+			int RecordID = m_logs[i].getRecord_ID();
+			ReportEngine re = null;
+			
+			re = ReportEngine.get (Env.getCtx(), reportEngineType, RecordID);
+			
+			pdfList.add(re.getPDF());				
+		}
+		if (pdfList.size() > 1) {
+			try {
+				File outFile = File.createTempFile(printTmpFileName, ".pdf");					
+				AEnv.mergePdf(pdfList, outFile);
+
+				Clients.clearBusy();
+				Window win = new SimplePDFViewer(getTitle(), new FileInputStream(outFile));
+				SessionManager.getAppDesktop().showWindow(win, "center");
+			} catch (Exception e) {
+				log.log(Level.SEVERE, e.getLocalizedMessage(), e);
+			}
+		} else if (pdfList.size() > 0) {
+			Clients.clearBusy();
+			try {
+				Window win = new SimplePDFViewer(getTitle(), new FileInputStream(pdfList.get(0)));
+				SessionManager.getAppDesktop().showWindow(win, "center");
+			} catch (Exception e)
+			{
+				log.log(Level.SEVERE, e.getLocalizedMessage(), e);
+			}
+		}
+		this.detach();
+	}
+	
+	/**
+	 * Is Printable
+	 * @param pi
+	 * @return
+	 */
+	public boolean isPrintable(ProcessInfo pi) {
+		for(ProcessInfoLog log : m_logs)
+			for(int tableID : ReportEngine.getDocTableIDs())
+				if(log.getAD_Table_ID() == tableID) {
+					reportEngineType = ReportEngine.getReportEngineType(tableID);

```

Comment:
should check reportEngineType is valie (i.e not -1) here.

Reply:
this loop of ReportEngine.getDocTableIDs() is not good, it should be simplify to just `if (ReportEngine.getReportEngineType(tableID) >= 0) return true;`